packages/ui/src/components/ChatList.tsx

```diff
@@ -242,164 +150,273 @@ export function ChatList({
+        duration: 200,
+      });
+    }
+  }, [isOpen, openProgress]);
 
-  const listStyle = useAnimatedStyle(() => {
+  const containerStyle = useAnimatedStyle(() => {
     return {
-      transform: [{ translateY: filterVisible.value ? FILTER_HEIGHT : 0 }],
+      overflow: 'hidden',
+      height: contentHeight * openProgress.value,
+      opacity: openProgress.value,
     };
   });
 
-  const scrollHandler = useAnimatedScrollHandler({
-    onScroll: (event) => {
-      if (!filterVisible.value) {
-        scrollY.value = event.contentOffset.y;
-      }
-    },
-    onEndDrag: (event) => {
-      if (event.contentOffset.y < -SNAP_THRESHOLD && !filterVisible.value) {
-        filterVisible.value = true;
-        scrollY.value = withSpring(-FILTER_HEIGHT);
-      } else if (
-        event.contentOffset.y > -SNAP_THRESHOLD &&
-        filterVisible.value
-      ) {
-        filterVisible.value = false;
-        scrollY.value = withSpring(0);
-      }
-    },
-  });
+  const handleContentLayout = useCallback((e: LayoutChangeEvent) => {
+    setContentHeight(e.nativeEvent.layout.height);
+  }, []);
 
-  useEffect(() => {
-    if (showFilters) {
-      filterVisible.value = true;
-      scrollY.value = withSpring(-FILTER_HEIGHT);
+  return (
+    <Animated.View style={containerStyle}>
+      <YStack
+        onLayout={handleContentLayout}
+        flexShrink={0}
+        backgroundColor="$background"
+        gap="$m"
+        position="absolute"
+        top={0}
+        left={0}
+        right={0}
+      >
+        <View paddingHorizontal="$l">
+          <Input>
+            <Input.Icon>
+              <Icon type="Search" />
+            </Input.Icon>
+            <Input.Area
+              value={query}
+              onChangeText={onQueryChange}
+              placeholder="Find by name"
+              spellCheck={false}
+              autoCorrect={false}
+              autoCapitalize="none"
+            />
+          </Input>
+        </View>
+        <Tabs>
+          <Tabs.Tab name="all" activeTab={activeTab} onTabPress={onPressTab}>
+            <Tabs.Title active={activeTab === 'all'}>All</Tabs.Title>
+          </Tabs.Tab>
+          <Tabs.Tab name="groups" activeTab={activeTab} onTabPress={onPressTab}>
+            <Tabs.Title active={activeTab === 'groups'}>Groups</Tabs.Title>
+          </Tabs.Tab>
+          <Tabs.Tab
+            name="messages"
+            activeTab={activeTab}
+            onTabPress={onPressTab}
+          >
+            <Tabs.Title active={activeTab === 'messages'}>Messages</Tabs.Title>
+          </Tabs.Tab>
+        </Tabs>
+      </YStack>
+    </Animated.View>
+  );
+}
+
+function useFilteredChats({
+  pinned,
+  unpinned,
+  pending,
+  searchQuery,
+  activeTab,
+}: {
+  pinned: Chat[];
+  unpinned: Chat[];
+  pending: Chat[];
+  searchQuery: string;
+  activeTab: TabName;
+}) {
+  const performSearch = useChatSearch({ pinned, unpinned });
+  const debouncedQuery = useDebouncedValue(searchQuery, 200);
```

Comment:
this seems like a great use case for `useTransition`, if you feel like it! could probably get snappier performance.